streams/src/test/java/org/apache/kafka/streams/processor/internals/StreamTaskTest.java

```diff
@@ -477,9 +465,19 @@ public boolean persistent() {
         final MockTime time = new MockTime();
         final StreamsConfig config = createConfig(baseDir);
         final StreamTask streamTask = new StreamTask(taskId, "appId", partitions, topology, consumer,
-                                                     changelogReader, config, streamsMetrics,
+                                                     changelogReader, producer, config, streamsMetrics,
                                                      stateDirectory, new ThreadCache("testCache", 0, streamsMetrics),
-                                                     time, recordCollector);
+                                                     time);
+
+        final Field taskRecordCollector = streamTask.getClass().getDeclaredField("recordCollector");
```

Comment:
This is a big code smell IMO. Using reflection to access fields in a class you are testing should be telling you something about the code. I'd prefer to see the `RecordCollector` back to being a ctor argument.